### PR TITLE
Update recipe and feedstock

### PR DESCRIPTION
**Pull request by jdblischak**
* Removed redundant run requirements (that are handled by run exports)
* Fixed linux-aarch64 builds. No binaries have been uploaded because it was configured to run on Travis, which isn't setup for this repository. I switched the builds to Azure
* Fixed the links to the Azure builds in the README
* Various other minor changes

Didn't bump the build number. None of these changes should have any effect on the binaries. New linux-aarch64 binaries will be uploaded though since they never have been before.

### CODE REVIEW

File: conda-forge.yml [hunk 1]
```diff
@@ -1,13 +1,17 @@
+azure:
+  build_id: 48
+  user_or_org: TileDB-Inc
+  project_name: CI
```
Comment (jdblischak) [inline conda-forge.yml:4]: This fixes the links to the Azure builds in the README

File: conda-forge.yml [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 build_platform:
   osx_arm64: osx_64
 conda_forge_output_validation: true
 github:
   branch_name: main
   tooling_branch_name: main
 provider:
-  linux_aarch64: default
+  linux_aarch64: azure
```
Comment (jdblischak) [inline conda-forge.yml:12]: This fixes the linux-aarch64 builds. The "default" option is Travis, which isn't enabled for this repository

**Review by teo-tsirpanis — APPROVED** [review]

File: conda-forge.yml [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
   win: azure
-test_on_native_only: true
+test: native_and_emulated
```
Comment (jdblischak) [inline conda-forge.yml:14]: Fixes a conda-smithy warning:

````
WARNING:conda_smithy.configure_feedstock:conda-forge.yml: $.test_on_native_only = True -> 'Test On Native Only' is deprecated.
This was used for disabling testing for cross-compiling.

```warning
This has been deprecated in favor of the top-level `test` field.
It is now mapped to `test: native_and_emulated`.
```
````

File: recipe/conda_build_config.yaml [hunk 4]
```diff
@@ -1,16 +1,13 @@
 MACOSX_SDK_VERSION:  # [osx and x86_64]
   - 10.15            # [osx and x86_64]
-MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
-  - "10.15"                # [osx and x86_64]
+c_stdlib_version:    # [osx and x86_64]
+  - "10.15"          # [osx and x86_64]
```
Comment (jdblischak) [inline recipe/conda_build_config.yaml:4]: Fixes a conda smithy warning:

```
WARNING:conda_smithy.configure_feedstock:Conflicting specification for minimum macOS deployment target!
If your conda_build_config.yaml sets `MACOSX_DEPLOYMENT_TARGET`, please change the name of that key to `c_stdlib_version`!

Using 10.15=max(c_stdlib_version, MACOSX_DEPLOYMENT_TARGET).
```

File: recipe/conda_build_config.yaml [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
 channel_targets:
   - tiledb main
 channel_sources:
   - conda-forge
 channel_priority:
   - strict
-tiledb:
-- '2.15.*'
-- '2.16.*'
```
Comment (jdblischak) [inline recipe/conda_build_config.yaml:13]: These were no longer have any effect. We pin `tiledb` in `meta.yaml`, and this overrides any setting in `conda_build_config.yaml`

File: recipe/meta.yaml [hunk 6]
```diff
@@ -45,11 +45,8 @@ requirements:
     - openblas
   run:
     - python
-    - {{ pin_compatible('numpy') }}
```
Comment (jdblischak) [inline recipe/meta.yaml:48]: numpy has switched to using run exports (and it has been for 2 years)

xref: [numpy2 migration announcement](https://conda-forge.org/news/2024/05/15/numpy-2-migration/), https://github.com/conda-forge/tiledb-py-feedstock/pull/218

File: recipe/meta.yaml [hunk 7]
```diff
@@ -45,11 +45,8 @@ requirements:
...
     - scikit-learn
-    - tiledb >=2.24,<2.25
```
Comment (jdblischak) [inline recipe/meta.yaml:50]: Handled by run exports

File: recipe/meta.yaml [hunk 8]
```diff
@@ -45,11 +45,8 @@ requirements:
...
     - tiledb-py
-    - libopenblas
```
Comment (jdblischak) [inline recipe/meta.yaml:52]: Handled by run exports

File: conda-forge.yml (hunk 1 above)
Comment (jdblischak) [reply conda-forge.yml:4]: Note that this also requires that the pipeline name be exactly "tiledb-vector-search-feedstock". The default name of a GitHub repo connected to azure includes the org name, in this case "TileDB-Inc.tiledb-vector-search-feedstock". I already renamed it

image

**Pull request merged**